### ISSUE TITLE
Sub item doesn't deselect if the list was collapsed when new click performed

### ISSUE THREAD

**Issue by masterleclerc** [mikepenz/MaterialDrawer#2184]
If the sub item was selected, and then we collapse the list and click on the other main item, both items will be selected (the new one and the old sub item)

How to reproduce (sample app):
1. Open navigation drawer
2. Expand "Collapsable Badge", select the first item - "CollapsableItem"
3. Collapse the list by clicking on "Collapsable Badge"
4. Click on "Contract"
5. Expand "Collapsable Badge"
6. Both "CollapsableItem" and "Contract" item will be selected

My temporary workaround - I deselect all sub items when the click was performed on the main items:

```
             .withOnDrawerItemClickListener(new Drawer.OnDrawerItemClickListener() {
                    @Override
                    public boolean onItemClick(View view, int position, IDrawerItem drawerItem) {
                        if (nonNull(drawerItem) && drawerItem instanceof AbstractBadgeableDrawerItem) {
                            long clickedIdentifier = drawerItem.getIdentifier();
                            if (!isSubItem(clickedIdentifier)) {
                                deselectAllSubItems();
                            }
                            (...)
                        }
                        return false;
                    }

   private void deselectAllSubItems() {
        List<IDrawerItem> drawerItems = drawerResult.getDrawerItems();
        for (IDrawerItem drawerItem : drawerItems) {
            if (drawerItem instanceof ExpandableDrawerItem) {
                ExpandableDrawerItem expandableDrawerItem = (ExpandableDrawerItem) drawerItem;
                List<IDrawerItem> subItems = expandableDrawerItem.getSubItems();
                if (nonNull(subItems)) {
                    for (IDrawerItem subItem : subItems) {
                        subItem.withSetSelected(false);
                    }
                }
            }
        }
    }

   private boolean isSubItem(long itemId) {
        List<IDrawerItem> drawerItems = drawerResult.getDrawerItems();
        for (IDrawerItem drawerItem : drawerItems) {
            if (drawerItem instanceof ExpandableDrawerItem) {
                ExpandableDrawerItem expandableDrawerItem = (ExpandableDrawerItem) drawerItem;
                List<IDrawerItem> subItems = expandableDrawerItem.getSubItems();
                if (nonNull(subItems)) {
                    for (IDrawerItem subItem : subItems) {
                        if (subItem.getIdentifier() == itemId)
                            return true;
                    }
                }
            }
        }
        return false;
    }
```

**Comment by mikepenz** [comment]
@masterleclerc there will be some enhancements in the next update (via a new `FastAdapter` version) to expandables. Please stay tuned ;)

**Comment by mikepenz** [comment]
Should be solved in the latest release v6.0.3 with the latest FastAdapter 3.2.1

**Comment by masterleclerc** [comment]
I can still reproduce the problem in v6.0.4
